
Name:

Institution: 

Date: 

### Detecting Parkinson’s Disease using Data-Driven Classification Model

##### What is Parkinson’s Disease?
Parkinson's disease is a progressive disorder of the central nervous system affecting movement and inducing tremors and stiffness . 

It has 5 stages to it and affects more than 1 million Individuals every year in India.
This is chronic and has no cure yet. It is a neurodegenerative disorder affecting dopamine-producing neurons in the brain. 
The aim of the project is to Detect Parkinson's Disease.

In this Python machine learning project I will leverage on the Python libraries scikit-learn, numpy, pandas, and xgboost, to build a model using an XGBClassifier. I'll start by loading the  the parkinsons dataset from  the UCI archive, get the features and labels, scale the features, then split the dataset, build an XGBClassifier, and then calculate the accuracy of the model.

#### Importing the necessary libraries

In [25]:
import numpy as np
import pandas as pd
import os, sys
from sklearn.preprocessing import MinMaxScaler
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

#### Loading the Parkinsons Dataset

In [26]:
df=pd.read_csv('parkinsons.data.csv')
df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)  \
0  phon_R01_S01_1      119.992       157.302        74.997         0.00784   
1  phon_R01_S01_2      122.400       148.650       113.819         0.00968   
2  phon_R01_S01_3      116.682       131.111       111.555         0.01050   
3  phon_R01_S01_4      116.676       137.871       111.366         0.00997   
4  phon_R01_S01_5      116.014       141.781       110.655         0.01284   

   MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...  \
0           0.00007   0.00370   0.00554     0.01109       0.04374  ...   
1           0.00008   0.00465   0.00696     0.01394       0.06134  ...   
2           0.00009   0.00544   0.00781     0.01633       0.05233  ...   
3           0.00009   0.00502   0.00698     0.01505       0.05492  ...   
4           0.00011   0.00655   0.00908     0.01966       0.06425  ...   

   Shimmer:DDA      NHR     HNR  status      RPDE       DFA   spread1  \
0      0.06545  0.02211  21.033       1  0.414783  0.815285 -4.813031   
1      0.09403  0.01929  19.085       1  0.458359  0.819521 -4.075192   
2      0.08270  0.01309  20.651       1  0.429895  0.825288 -4.443179   
3      0.08771  0.01353  20.644       1  0.434969  0.819235 -4.117501   
4      0.10470  0.01767  19.649       1  0.417356  0.823484 -3.747787   

    spread2        D2       PPE  
0  0.266482  2.301442  0.284654  
1  0.335590  2.486855  0.368674  
2  0.311173  2.342259  0.332634  
3  0.334147  2.405554  0.368975  
4  0.234513  2.332180  0.410335  

[5 rows x 24 columns]

#### Exploring the dataset

In [27]:
df.shape

(195, 24)

##### checking for any missing values in the data

In [28]:
df.isnull().sum()

name                0
MDVP:Fo(Hz)         0
MDVP:Fhi(Hz)        0
MDVP:Flo(Hz)        0
MDVP:Jitter(%)      0
MDVP:Jitter(Abs)    0
MDVP:RAP            0
MDVP:PPQ            0
Jitter:DDP          0
MDVP:Shimmer        0
MDVP:Shimmer(dB)    0
Shimmer:APQ3        0
Shimmer:APQ5        0
MDVP:APQ            0
Shimmer:DDA         0
NHR                 0
HNR                 0
status              0
RPDE                0
DFA                 0
spread1             0
spread2             0
D2                  0
PPE                 0
dtype: int64

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 24 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              195 non-null    object 
 1   MDVP:Fo(Hz)       195 non-null    float64
 2   MDVP:Fhi(Hz)      195 non-null    float64
 3   MDVP:Flo(Hz)      195 non-null    float64
 4   MDVP:Jitter(%)    195 non-null    float64
 5   MDVP:Jitter(Abs)  195 non-null    float64
 6   MDVP:RAP          195 non-null    float64
 7   MDVP:PPQ          195 non-null    float64
 8   Jitter:DDP        195 non-null    float64
 9   MDVP:Shimmer      195 non-null    float64
 10  MDVP:Shimmer(dB)  195 non-null    float64
 11  Shimmer:APQ3      195 non-null    float64
 12  Shimmer:APQ5      195 non-null    float64
 13  MDVP:APQ          195 non-null    float64
 14  Shimmer:DDA       195 non-null    float64
 15  NHR               195 non-null    float64
 16  HNR               195 non-null    float64
 1

##### Observations

There are in total 195 samples in the data set

The dataset contains 24 columns and 195 rows

The are no missing values in the dataset

##### Avarage number per column

In [30]:
df.groupby('status').mean()

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)  \
status                                                            
0        181.937771    223.636750    145.207292        0.003866   
1        145.180762    188.441463    106.893558        0.006989   

        MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  \
status                                                                   
0               0.000023  0.001925  0.002056    0.005776      0.017615   
1               0.000051  0.003757  0.003900    0.011273      0.033658   

        MDVP:Shimmer(dB)  ...  MDVP:APQ  Shimmer:DDA       NHR        HNR  \
status                    ...                                               
0               0.162958  ...  0.013305     0.028511  0.011483  24.678750   
1               0.321204  ...  0.027600     0.053027  0.029211  20.974048   

            RPDE       DFA   spread1   spread2        D2       PPE  
status                                                              
0       0.442552  0.695716 -6.759264  0.160292  2.154491  0.123017  
1       0.516816  0.725408 -5.333420  0.248133  2.456058  0.233828  

[2 rows x 22 columns]

#### Preparing the dataset for modelling

###### Getting the features and labels from the Dataset

The features are all the columns/variables except ‘status’ and the labels are those in the ‘status’ column


In [31]:
features = df.drop(['name', 'status'], axis=1)
labels = df.status

#### Scaling the features in our dataset

It Involves initializing a MinMaxScaler and scale the features to between -1 and 1 to normalize them. 

The MinMaxScaler transforms features by scaling them to a given range. 

The fit_transform() method fits to the data and then transforms it. 

In [32]:
scaler=MinMaxScaler((-1,1))
x=scaler.fit_transform(features)
y=labels

#### Performing a Train Test Split on the Data¶

In [41]:
x_train,x_test,y_train,y_test=train_test_split(x, y, test_size=0.2, random_state=10)

#### Performing our model using Model using  XGBClassifier   

This classifies using eXtreme Gradient Boosting- using gradient boosting algorithms for modern data science problems.

It falls under the category of Ensemble Learning in ML, where we train and predict using many models to produce one superior output.

In [42]:
model=XGBClassifier()

#### Trainning our model on the data

In [43]:
model.fit(x_train,y_train)

[18:15:11] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.3.0/src/learner.cc:1061: Starting in XGBoost 1.3.0, the default evaluation metric used with the objective 'binary:logistic' was changed from 'error' to 'logloss'. Explicitly set eval_metric if you'd like to restore the old behavior.


XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
       colsample_bynode=1, colsample_bytree=1, gamma=0, gpu_id=-1,
       importance_type='gain', interaction_constraints='',
       learning_rate=0.300000012, max_delta_step=0, max_depth=6,
       min_child_weight=1, missing=nan, monotone_constraints='()',
       n_estimators=100, n_jobs=4, num_parallel_tree=1,
       objective='binary:logistic', random_state=0, reg_alpha=0,
       reg_lambda=1, scale_pos_weight=1, subsample=1, tree_method='exact',
       use_label_encoder=True, validate_parameters=1, verbosity=None)

#### calculating the accuracy of the model

In [44]:
y_pred=model.predict(x_test)
print(accuracy_score(y_test, y_pred)*100)

100.0


#### Summary

In this machine machine learning analysis, I have build a model with a capability to detect the presence of Parkinson’s Disease in individuals using various factors.

I have used an XGBClassifier for this and made use of the sklearn library to prepare the dataset. 

This gives us an accuracy of 100%, which is great considering the number of lines of code and the size of the dataset in this python project.